pkg/lwip/contrib/sock/lwip_sock.c

```diff
@@ -616,22 +598,20 @@ ssize_t lwip_sock_sendv(struct netconn *conn, sock_tx_snip_t *snips,
         }
     }
 
-    for (sock_tx_snip_t *snip = snips; snip != NULL; snip = snip->next) {
-        buf = netbuf_new();
-        if ((buf == NULL) || (netbuf_alloc(buf, snip->len) == NULL)) {
+    for (const sock_tx_snip_t *snip = snips; snip != NULL; snip = snip->next) {
+        struct netbuf *tail = netbuf_new();
```

Comment:
seeing this. is the data copied? if that's the case, we could allocate the whole pkt, then write the snips in there. saves allocations and thus overhead.

Reply:
tbh this surprised me, doesn't LWIP support the concept of memory chunks? It certainly does for RX - maybe @miri64 or @yarrick know how to do this without copying the data

Reply:
see 1bad89cf6194a1a34ef66181df6b57a0792e34fa